core/src/main/scala/kafka/server/BrokerToControllerChannelManagerImpl.scala

```diff
@@ -125,15 +126,25 @@ class BrokerToControllerChannelManagerImpl(metadataCache: kafka.server.MetadataC
   }
 
   override def sendRequest(request: AbstractRequest.Builder[_ <: AbstractRequest],
-                           callback: RequestCompletionHandler): Unit = {
-    requestQueue.put(BrokerToControllerQueueItem(request, callback))
+                           callback: BrokerToControllerRequestCompletionHandler,
+                           requestTimeout: Long): Unit = {
+    requestQueue.put(BrokerToControllerQueueItem(request, callback, time.milliseconds() + requestTimeout))
```

Comment:
Won't this overflow with `requestTimeout` set to `Long.MaxValue`. Do we have any test cases?

Reply:
Good catch!